### PR TITLE
Transaction Mode and VSchema e2e GO test cases

### PR DESCRIPTION
**Pull request by ajeetj**
Transaction Mode and VSchema e2e GO test cases migrated from python and written using new GO cluster.

Signed-off-by: Ajeet jain <ajeet@planetscale.com>

### CODE REVIEW

**Comment by ajeetj** [comment]
@deepthi 
1. Created a base/master branch for all go tests.
2. Using that branch as base wrote Transaction mode and Vschema test cases.

Tests are the same which you have previously reviewed, the major changes are in TestMain function.

File: go/test/endtoend/vtgate/transaction/trxn_mode_test.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	exitcode, err := func() (int, error) {
+		clusterInstance = &cluster.LocalProcessCluster{Cell: cell, Hostname: hostname}
+		defer clusterInstance.Teardown()
+
+		// Reserver vtGate port inorder to pass it to vtTablet
```
Comment (deepthi) [inline go/test/endtoend/vtgate/transaction/trxn_mode_test.go]: Typos:
Reserver -> Reserve
inorder -> in order

File: go/test/endtoend/vtgate/transaction/trxn_mode_test.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	got := fmt.Sprintf("%v", qr.Rows)
+	want = `[[INT64(3)]]`
+	assert.Equal(t, want, got)
+
+	qr = exec(t, conn2, "select name from twopc_lookup where id=3")
+	got = fmt.Sprintf("%v", qr.Rows)
+	want = `[[VARCHAR("mark")]]`
+	assert.Equal(t, want, got)
+
+	// DELETE from multiple tables using TWOPC trnx mode
```
Comment (deepthi) [inline go/test/endtoend/vtgate/transaction/trxn_mode_test.go]: trnx -> transaction

File: go/test/endtoend/vtgate/vschema/vschema_test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+func TestVSchema(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	// Test the blank database with no vschema
```
Comment (deepthi) [inline go/test/endtoend/vtgate/vschema/vschema_test.go]: blank -> empty

File: go/test/endtoend/vtgate/vschema/vschema_test.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	got := fmt.Sprintf("%v", qr.Rows)
+	want := `[[INT64(1) VARCHAR("test1")] [INT64(2) VARCHAR("test2")] [INT64(3) VARCHAR("test3")] [INT64(4) VARCHAR("test4")]]`
+	assert.Equal(t, want, got)
+
+	qr = exec(t, conn, "delete from vt_user")
+	got = fmt.Sprintf("%v", qr.Rows)
+	want = `[]`
+	assert.Equal(t, want, got)
+
+	// Test Blank VSCHEMA
```
Comment (deepthi) [inline go/test/endtoend/vtgate/vschema/vschema_test.go]: Blank -> empty
VSCHEMA -> vschema

**Review by deepthi — APPROVED** [review]
Looks good. Just a few typos / word changes in comments.
You can merge after fixing them.

**Pull request merged**